pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -65,6 +65,7 @@ const (
 // AttachDetachController defines the operations supported by this controller.
 type AttachDetachController interface {
 	Run(stopCh <-chan struct{})
+	GetDesiredStateOfWorld() cache.DesiredStateOfWorld
```

Comment:
`adc` is not exposed from the package and hence can't be read from the test directly. :( 
That is why we are left with only option of having this function in the interface.  Overall, though I think it should be okay.